### ISSUE TITLE
Allow new Quick APIs `aroundEach` and `justBeforeEach`

### ISSUE THREAD

**Issue by stonko1994** [realm/SwiftLint#4626]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Quick [`5.0.0`](https://github.com/Quick/Quick/releases/tag/v5.0.0) introduced `aroundEach` and Quick [`6.0.0`](https://github.com/Quick/Quick/releases/tag/v6.0.0) introduced `justBeforeEach`. Both are currently flagged as violations for `quick_discouraged_call`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
⚠️ Line xxx: Discouraged call inside a 'describe' block.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.50.1`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `Homebrew `
* Paste your configuration file: `N/A`

```swift
// This triggers a violation:
describe("foo") {
    justBeforeEach {
        ...
    }
}
```
